### PR TITLE
Update the jmx-metrics version to 1.17.0-alpha

### PR DESCRIPTION
**Pull request by opentelemetrybot**
Update the jmx-metrics version to `1.17.0-alpha`.

### CODE REVIEW

File: receiver/jmxreceiver/supported_jars.go [hunk 1]
```diff
@@ -42,6 +42,10 @@ func oldFormatProperties(c *Config, j supportedJar) error {
 // If you change this variable name, please open an issue in opentelemetry-java-contrib
 // so that repository's release automation can be updated
 var jmxMetricsGathererVersions = map[string]supportedJar{
+	"2ce5255ee07e139b93011f780d814dbe9440bf1eded35f6bd5c7776dbc3fcd70": {
+		version: "1.17.0-alpha",
```
Comment (trask) [inline receiver/jmxreceiver/supported_jars.go:46]: I noticed the versions below don't have `-alpha`, so maybe the `-alpha` was being added somewhere else, in which case maybe this is breaking and will look for `1.17.0-alpha-alpha`?

**Comment by djaglowski** [comment]
cc: @dehaansa @Mrod1598

File: receiver/jmxreceiver/supported_jars.go (hunk 1 above)
Comment (dehaansa) [reply receiver/jmxreceiver/supported_jars.go:46]: That particular version field is currently primarily maintainer documentation of the currently supported versions, and not enforced in any way. The only thing being enforced is that the hash of the provided jar matches something in the list.

**Review by dehaansa — APPROVED** [review]

File: receiver/jmxreceiver/supported_jars.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/jmxreceiver/supported_jars.go:46]: @dehaansa let's be consistent. use `-alpha` everywhere or nowhere :)

Comment (dehaansa) [reply receiver/jmxreceiver/supported_jars.go:46]: Added this and other missing versions (and added `-alpha`) in new PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14696

**Comment by djaglowski** [comment]
Who manages the bot that created this PR? Can we have it add a changelog entry for future PRs?

**Comment by trask** [comment]
> Who manages the bot that created this PR? Can we have it add a changelog entry for future PRs?

hey! it's coming from https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/.github/workflows/reusable-create-collector-contrib-pull-request.yml

I will update it to add a changelog entry 👍

**Comment by dehaansa** [comment]
This particular PR should probably close in favor of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14696 , I updated the older versions to match the `-alpha` pattern as requested and added the newer versions, too.

**Pull request closed without merging**